### PR TITLE
preview v0.1.3 blog post

### PR DESCRIPTION
**Pull request by jldec**

### CODE REVIEW

File: _posts/2018-09-19-announcing-riff-0-1-3-on-Knative.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+header:
+  overlay_image: /images/cam-bridge.jpg
+excerpt:
+  With a riff buildpack for java, credential helpers, content-type shortcuts, and subscription commands
+permalink: /blog/announcing-riff-0-1-3-on-knative/
+---
+
+We are pleased to announce the release of [riff v0.1.3](https://github.com/projectriff/riff/releases/tag/v0.1.3) on Knative. Thank you riff and Knative contributors.
+
+The riff CLI can be downloaded from our [releases page](https://github.com/projectriff/riff/releases/tag/v0.1.2) on GitHub. Please follow one of the [getting started](/docs) guides, to create a new cluster on GKE or minikube. This release includes new manifests for the latest Knative and Istio.
```
Comment (ericbottard) [inline _posts/2018-09-19-announcing-riff-0-1-3-on-Knative.md:13]: This references the 0.1.2 files, make sure to change before publish

File: _posts/2018-09-19-announcing-riff-0-1-3-on-Knative.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+riff namespace init default --gcr <path-to-json-file>
+```
+
+#### or `export DOCKER_ID=<your-id>` and enter password when prompted
+```sh
+riff namespace init default --dockerhub $DOCKER_ID
+```
+
+## riff buildpack for java
+This release of riff supports building Java functions from source, either locally or in the cloud. Both variants use a new riff buildpack for Java.
```
Comment (ericbottard) [inline _posts/2018-09-19-announcing-riff-0-1-3-on-Knative.md:39]: should "in the cloud" rather say "on the cluster"?

File: _posts/2018-09-19-announcing-riff-0-1-3-on-Knative.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+All you need in your directory is the java code with a maven pom, and the name of the handler class in a file called 'riff.toml'. The example below uses a sample [hello-java](https://github.com/jldec/hello-java) function on GitHub.
```
Comment (ericbottard) [inline _posts/2018-09-19-announcing-riff-0-1-3-on-Knative.md:41]: riff.toml should use backticks maybe?

Comment (ericbottard) [reply _posts/2018-09-19-announcing-riff-0-1-3-on-Knative.md:41]: Should this read *available* on Github? The phrase "on github" seems to imply that you're directly building from there IMO (ie `--git-repo`)

File: _posts/2018-09-19-announcing-riff-0-1-3-on-Knative.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+#### build from code in a directory and push to local docker
+```sh
+riff function create java-buildpack hello --local-path . --image dev.local/hello-java:v1
+```
+Local builds run directly on your machine, not inside the Kubernetes cluster. The `dev.local` tag, sends the image to your local docker daemon.
```
Comment (ericbottard) [inline _posts/2018-09-19-announcing-riff-0-1-3-on-Knative.md:47]: Another case where there may be a confusion: local builds don't mandate to just use the local build daemon. You can do local build AND push to a remote repository.
Also, should not use the word "tag" to refer to `dev.local` IMO

File: _posts/2018-09-19-announcing-riff-0-1-3-on-Knative.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+riff function create java-buildpack hello --local-path . --image dev.local/hello-java:v1
+```
+Local builds run directly on your machine, not inside the Kubernetes cluster. The `dev.local` tag, sends the image to your local docker daemon. 
+
+
+#### build from code on GitHub and push to DockerHub 
+```sh
+riff function create java-buildpack hello --git-repo https://github.com/jldec/hello-java.git --image $DOCKER_ID/hello-java:v1 --verbose
+```
+Using --verbose shows the progress of the build as it's happening in the cluster. For GCR, replace $DOCKER_ID with your gcr.io/$GCP_PROJECT.
```
Comment (ericbottard) [inline _posts/2018-09-19-announcing-riff-0-1-3-on-Knative.md:54]: This sentence could use some backticks love

Comment (ericbottard) [reply _posts/2018-09-19-announcing-riff-0-1-3-on-Knative.md:54]: This sentence, being the sole sentence, feels like it is the explanation of the "build from code on GitHub and push to DockerHub", but it isn't. Maybe a first sentence focusing on the use of `git-repo` should be added

File: _posts/2018-09-19-announcing-riff-0-1-3-on-Knative.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+#### build from code on GitHub and push to DockerHub 
+```sh
+riff function create java-buildpack hello --git-repo https://github.com/jldec/hello-java.git --image $DOCKER_ID/hello-java:v1 --verbose
+```
+Using --verbose shows the progress of the build as it's happening in the cluster. For GCR, replace $DOCKER_ID with your gcr.io/$GCP_PROJECT. 
+
+
+## simpler riff service invoke 
+For text or json, you can now use --text or --json. You no longer have to specify the Content-Type as a curl header.
```
Comment (ericbottard) [inline _posts/2018-09-19-announcing-riff-0-1-3-on-Knative.md:58]: Add backticks to `--text` and `--json` (maybe even `Content-Type`)

File: _posts/2018-09-19-announcing-riff-0-1-3-on-Knative.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+#### invoke the hello function with text input
+```sh
+riff service invoke hello --text -- -w '\n' -d world
+```
+
+## riff subscription commands
+Subscriptions now have their own separate commands in the riff CLI.
```
Comment (ericbottard) [inline _posts/2018-09-19-announcing-riff-0-1-3-on-Knative.md:66]: Should we state that support as a sideband of `riff service|function create` is removed?

Comment (ericbottard) [inline _posts/2018-09-19-announcing-riff-0-1-3-on-Knative.md:66]: Also, seems like we'll have `riff subscription list` in time for the release

File: _posts/2018-09-19-announcing-riff-0-1-3-on-Knative.md (hunk 3 above)
Comment (jldec) [reply _posts/2018-09-19-announcing-riff-0-1-3-on-Knative.md:41]: i tried backticks inline but this theme doesn't have good formatting for that and it just looks a bit off.

Comment (jldec) [reply _posts/2018-09-19-announcing-riff-0-1-3-on-Knative.md:41]: i fiddled with the style for backticks inline. looks a bit better now.

File: _posts/2018-09-19-announcing-riff-0-1-3-on-Knative.md (hunk 5 above)
Comment (jldec) [reply _posts/2018-09-19-announcing-riff-0-1-3-on-Knative.md:54]: i think it's ok with the header above the code block.

**Pull request merged**